solutions/USACO jan17-cownav.mdx

```diff
@@ -38,6 +38,157 @@ Don't forget that once Bessie reaches the goal, she will ignore further commands
 
 ### Implementation
 
-<!-- I really don't want to write Cow Navigation. Anyone have code? -->
 
-<IncompleteSection />
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+//created by Oleksandr Gorpynich
+
+//This implementation also keeps track of whether cow 1 (facing left) and cow 2 (facing up) 
+//have reached the end at some point. This adds a factor of 4 to the solution however it still
+//runs fast enough
+#pragma GCC optimize("O2")
+#include <fstream>
+#include <string>
+#include <iostream>
+#include <bitset>
+#include <math.h>
+#include <string>
+#include <algorithm>
+#include <assert.h>
+#include<bits/stdc++.h>
+#include <vector>
+#include <queue>
+#include<stdio.h>
+#include<ctype.h>
+#define ll long long
+using namespace std;
+
+int n;
+
+struct state{
+    int x1;
+    int y1;
+    int dir1;
+    int x2;
+    int y2;
+    int dir2;
+    int finished1;
+    int finished2;
+    int dist;
+};
+char arr[21][21];
+int visited[21][21][4][21][21][4][2][2];
```

Comment:
You don't actually need this many states; you only care about the two starting states and the "net" direction. When either starting state reaches the end, simply choose to ignore its movement. 

\# of states can be reduced to 20 x 20 x 20 x 20 x 4

Reply:
Whether you want to change this is up to you

Reply:
Also you can turn visited into a boolean array

Reply:
ok sounds good, I will fix it

Reply:
Ok, It has been fixed. I didn't know about the memset function which makes it so much cleaner so thanks for that.

Reply:
You're welcome!